torch/_inductor/codegen/triton.py

```diff
@@ -139,6 +139,23 @@ def gen_common_triton_imports():
 }
 
 
+def get_symt(tree: IterationRangesEntry) -> SymT:
+    prefix_to_symt = {prefix: symt for symt, prefix in prefix_str.items()}
+    return prefix_to_symt[tree.prefix]
+
+
+def get_block_size(tree: IterationRangesEntry) -> sympy.Symbol:
+    return block_sizes[get_symt(tree)]
+
+
+def get_block_offset(tree: IterationRangesEntry) -> sympy.Symbol:
+    return block_offsets[get_symt(tree)]
+
+
+def max_block_size(tree: IterationRangesEntry) -> int:
+    return TRITON_MAX_BLOCK[tree.prefix.upper()]
```

Comment:
Why not adding those as methods for IterationRangeEntry class?

Also it would be good to precompute and store the result for get_symt (or add a cache).

Reply:
Good idea, I think this makes sense for `get_symt` at least. I'll try to refactor it and add caching.

For the others, my only concern is that they're Triton-specific concepts whereas `IterationRangesEntry` is generic to other backends.

Reply:
In [72dabc7](https://github.com/pytorch/pytorch/pull/135557/commits/72dabc7fc9c84e21d2eba107b4cb04fe47f065b0), I refactored `get_symt` to be a cached method of `IterationRanges`.

Since the others are triton-specific, I cleaned them up a bit by refactoring them into the new `TritonSymbols` class.